api/pkg/transformer/feast/redis/client.go

```diff
@@ -0,0 +1,106 @@
+package redis
+
+import (
+	"context"
+	"time"
+
+	feast "github.com/feast-dev/feast/sdk/go"
+	"github.com/go-redis/redis/v8"
+	"github.com/gojek/merlin/pkg/transformer/spec"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+type RedisEncoder struct {
+	specs map[string]*spec.FeatureTableMetadata
+}
+
+type RedisClient struct {
+	encoder   RedisEncoder
+	pipeliner redis.Pipeliner
+}
+
+func NewRedisClient(redisStorage *spec.RedisStorage, featureTablesMetadata []*spec.FeatureTableMetadata) (RedisClient, error) {
+	option := redisStorage.Option
+	redisClient := redis.NewClient(&redis.Options{
+		Addr:               redisStorage.GetRedisAddress(),
+		MaxRetries:         int(option.MaxRetries),
+		MinRetryBackoff:    getNullableDuration(option.MinRetryBackoff),
+		MaxRetryBackoff:    getNullableDuration(option.MinRetryBackoff),
+		DialTimeout:        getNullableDuration(option.DialTimeout),
+		ReadTimeout:        getNullableDuration(option.ReadTimeout),
+		WriteTimeout:       getNullableDuration(option.WriteTimeout),
+		PoolSize:           int(option.PoolSize),
+		MaxConnAge:         getNullableDuration(option.MaxConnAge),
+		PoolTimeout:        getNullableDuration(option.PoolTimeout),
+		IdleTimeout:        getNullableDuration(option.IdleTimeout),
+		IdleCheckFrequency: getNullableDuration(option.IdleCheckFrequency),
+		MinIdleConns:       int(option.MinIdleConnections),
+	})
+
+	return RedisClient{
+		encoder:   newRedisEncoder(featureTablesMetadata),
+		pipeliner: redisClient.Pipeline(),
+	}, nil
+}
+
+func NewRedisClusterClient(redisClusterStorage *spec.RedisClusterStorage, featureTablesMetadata []*spec.FeatureTableMetadata) (RedisClient, error) {
+	option := redisClusterStorage.Option
+	redisClient := redis.NewClusterClient(&redis.ClusterOptions{
+		Addrs:              redisClusterStorage.GetRedisAddress(),
+		MaxRetries:         int(option.MaxRetries),
+		MinRetryBackoff:    getNullableDuration(option.MinRetryBackoff),
+		MaxRetryBackoff:    getNullableDuration(option.MinRetryBackoff),
+		DialTimeout:        getNullableDuration(option.DialTimeout),
+		ReadTimeout:        getNullableDuration(option.ReadTimeout),
+		WriteTimeout:       getNullableDuration(option.WriteTimeout),
+		PoolSize:           int(option.PoolSize),
+		MaxConnAge:         getNullableDuration(option.MaxConnAge),
+		PoolTimeout:        getNullableDuration(option.PoolTimeout),
+		IdleTimeout:        getNullableDuration(option.IdleTimeout),
+		IdleCheckFrequency: getNullableDuration(option.IdleCheckFrequency),
+		MinIdleConns:       int(option.MinIdleConnections),
+	})
+
+	return RedisClient{
+		encoder:   newRedisEncoder(featureTablesMetadata),
+		pipeliner: redisClient.Pipeline(),
+	}, nil
+}
+
+func (r RedisClient) GetOnlineFeatures(ctx context.Context, req *feast.OnlineFeaturesRequest) (*feast.OnlineFeaturesResponse, error) {
```

Comment:
Can you add test which involve GetOnlineFeatures? It seems that `client.go` is never tested.